File: doc/codes/python/examples/api/command/modulation.py
from autd3.commands import ChangeModulationBank, ConfigModulation, Modulation, WriteModulationBuffer
from autd3.value import LoopBehavior, ModulationBank, SamplingConfig, TransitionMode
from autd3_modulation import modulation_buffer

bank = ModulationBank.B0
config = SamplingConfig.FREQ_4K
loop_behavior = LoopBehavior.Infinite
transition_mode = TransitionMode.Immediate

data = modulation_buffer()

# ANCHOR: api
Modulation(config, data)

Modulation(
    bank=bank,
    config=config,
    data=data,
    loop_behavior=loop_behavior,
    transition_mode=transition_mode,
)
# ANCHOR_END: api

# ANCHOR: equivalent
WriteModulationBuffer(
    bank=bank,
    offset=0,
    data=data,
)
ConfigModulation(
    bank=bank,
    config=config,
    size=len(data),
    loop_behavior=loop_behavior,
)
ChangeModulationBank(
    bank=bank,
    transition_mode=transition_mode,
)
# ANCHOR_END: equivalent

File: doc/codes/csharp/examples/api/command/modulation.cs
using AUTD3;

namespace DocSamples.ApiCommandModulation;

internal static class Sample
{
    internal static void Run()
    {
        var bank = ModulationBank.B0;
        var config = SamplingConfig.Freq4k;
        var loopBehavior = LoopBehavior.Infinite;
        var transitionMode = TransitionMode.Immediate;

        var data = Modulation.ModulationBuffer();

        // ANCHOR: api
        new Modulation(config, data);

        new Modulation(bank, config, data);

        new Modulation(
            bank: bank,
            config: config,
            data: data,
            loopBehavior: loopBehavior,
            transitionMode: transitionMode
        );
        // ANCHOR_END: api

        // ANCHOR: equivalent
        new WriteModulationBuffer(
            bank: bank,
            offset: 0,
            data: data
        );
        new ConfigModulation(
            bank: bank,
            config: config,
            size: (uint)data.Length,
            loopBehavior: loopBehavior
        );
        new ChangeModulationBank(
            bank: bank,
            transitionMode: transitionMode
        );
        // ANCHOR_END: equivalent
    }
}
